### PR TITLE
[LS][SemanticAPI] Refactor Abstract Completion provider to include isolated variable symbols  conditionally

### LINKED ISSUE #30637: [LS] Isolated variables are not filtered based on the context when providing completions
**Description:**
Isolated variables(symbols) are currently suggested irrespective of whether the cursor is in lock statement context.

**Steps to reproduce:**
Try the following code segment in VSCode
```ballerina
isolated map<Coordinates> cities = {};

function add(string city, Coordinates coordinates) {
    lock {
        if cities.hasKey(city) {
            return;
        }

        cities[city] = coordinates.clone();
    }
    var myStr = <cursor>

}
```

### PR DESCRIPTION
**Pull request by malinthar**
## Purpose
$Subject. 
With this change variable completion items corresponding to isolated variable symbols will only be created when the cursor is positioned with in a lock statement.

Fixes #30637 #30672 

## Check List 
- [x] Read the [Contributing Guide](https://github.com/ballerina-platform/ballerina-lang/blob/master/CONTRIBUTING.md)
- [ ] Updated Change Log
- [ ] Checked Tooling Support (#<Issue Number>)
- [ ] Added necessary tests
  - [x] Unit Tests
  - [ ] Spec Conformance Tests
  - [ ] Integration Tests
  - [ ] Ballerina By Example Tests
- [x] Increased Test Coverage   
- [ ] Added necessary documentation  
  - [ ] API documentation 
  - [ ] Module documentation in Module.md files
  - [ ] Ballerina By Examples

### CODE REVIEW

File: language-server/modules/langserver-core/src/main/java/org/ballerinalang/langserver/common/utils/CommonUtil.java [hunk 1]
```diff
@@ -1222,4 +1223,23 @@ public static Boolean isInFunctionCallParameterContext(PositionedOperationContex
...
+        NonTerminalNode evalNode = context.getNodeAtCursor();
+        boolean withinLockStatementNode = false;
+        do {
+            if (evalNode.kind() == SyntaxKind.LOCK_STATEMENT) {
+                withinLockStatementNode = true;
+            }
+            evalNode = evalNode.parent();
+        }
+        while (!withinLockStatementNode && evalNode != null);
+        return withinLockStatementNode;
```
Comment (IMS94) [inline language-server/modules/langserver-core/src/main/java/org/ballerinalang/langserver/common/utils/CommonUtil.java:1243]: We don't need the `withinLockStatementNode` right?
```suggestion
        boolean withinLockStatementNode = false;
        do {
                    if (evalNode.kind() == SyntaxKind.LOCK_STATEMENT) {
                            return true;
                    }
                    evalNode = evalNode.parent();
       } while (evalNode != null);
       return false;
```

**Review by IMS94 — APPROVED** [review]
Approved with a minor comment.

File: language-server/modules/langserver-core/src/main/java/org/ballerinalang/langserver/common/utils/CommonUtil.java (hunk 1 above)
Comment (malinthar) [reply language-server/modules/langserver-core/src/main/java/org/ballerinalang/langserver/common/utils/CommonUtil.java:1243]: Agreed!

**Review by nadeeshaan — DISMISSED** [review]
LGTM

**Comment by mohanvive** [comment]
@malinthar can you please resolve the conflicts?

**Review by mohanvive — APPROVED** [review]

**Pull request merged**